src/Maui/Prism.Maui/Ioc/ContainerProvider.cs

```diff
@@ -0,0 +1,65 @@
+namespace Prism.IoC;
+
+/// <summary>
+/// Provides Types and Services registered with the Container
+/// </summary>
+/// <typeparam name="T">The type to Resolve</typeparam>
+/// <example>
+/// We can use this to build better types such as ValueConverters with full dependency injection
+/// <code>
+/// public class MyValueConverter : IValueConverter
+/// {
+///     private ILoggerFacade _logger { get; }
+///     public MyValueConverter(ILoggerFacade logger)
+///     {
+///         _logger = logger;
+///     }
+///
+///     public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+///     {
+///         _logger.Log($"Converting {value.GetType().Name} to {targetType.Name}", Category.Debug, Priority.None);
+///         // do stuff
+///     }
+///
+///     public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+///     {
+///         _logger.Log($"Converting back from {value.GetType().Name} to {targetType.Name}", Category.Debug, Priority.None);
+///         return null;
+///     }
+/// }
+/// </code>
+/// We can then simply use our ValueConverter or other class directly in XAML
+/// <![CDATA[
+/// <ContentPage xmlns:prism="clr-namespace:Prism.Ioc;assembly=Prism.Maui">
+///     <ContentPage.Resources>
+///         <ResourceDictionary>
+///             <prism:ContainerProvider x:TypeArguments="MyValueConverter" x:Key="myValueConverter" />
+///         <ResourceDictionary>
+///     <ContentPage.Resources>
+///     <Label Text="{Binding SomeProp, Converter={StaticResource myValueConverter}}" />
+/// </ContentPage>
+/// ]]>
+/// </example>
+public class ContainerProvider<T>
+{
+    /// <summary>
+    /// The Name used to register the type with the Container
+    /// </summary>
+    public string Name { get; set; }
```

Comment:
Thinking about nullable's, I wonder if the implicit operator should return `T?` instead?

There's also a null check on the container however I don't think it's needed as if `ContainerLocator.Container` is `null` it will throw an `InvalidOperationException` when being accessed.

Reply:
No need to return `T?` as the Container would throw an exception rather than return null.